### ISSUE TITLE
Using welcome-page to teach more

### ISSUE THREAD

**Issue by ef4** [ember-cli/ember-welcome-page#57]
Here is are some ideas for ember-welcome-page generated from an ember core team meeting today. We discussed simplifying the architecture so that:

 - the application blueprint would include an `application.hbs` file (because everyone needs to make one anyway, and providing a file ready-to-go is friendlier and less error prone).
 - the `application.hbs` blueprint would include an actual component that renders the welcome message, like:

    ```hbs
{{!-- The welcome-page component displays Ember's default welcome message. You should feel free to delete it. --}}
{{welcome-page}}
{{outlet}}
```
    The welcome message itself would invite the user to open `application.hbs` and make edits and see their results live reloaded in the browser. The benefit of this pattern is that users can immediately dive into the edit/test loop by poking at the template and seeing their changes taking effect.
 - This addon becomes simpler (no more middleware or shouldOverride checking). It would be a normal component-exporting addon, with one addition: it could use the `treeFor` APIs in `index.js` to only emit its code & assets in development builds (so that users who leave it in package.json don't end up with any unnecessary bytes).
 - future capabilities that have been discussed (like a built-in documentation browser) could be separate addons, we don't need to support those cases here.

**Comment by nathanhammond** [comment]
One note: it's not uncommon to find development builds published on the Internet. Are we comfortable with that tradeoff?

**Comment by ef4** [comment]
Including an extra handful of KB in an app that is already horribly under-optimized because it's built wrong seems like not a big risk.

**Comment by mike-north** [comment]
So much +1 to @ef4's response. We have the potential to be the most approachable SPA framework/library for beginners (by a huge margin), and moving in that direction at the expense of a few KB in improperly-configured builds (i.e. EMBER_ENV=dev for production) makes a lot of sense

**Comment by nathanhammond** [comment]
One additional feature request: bundled command in the welcome-page addon to remove the welcome-page addon (since we don't have generic uninstall this needs to be manually written).

**Comment by locks** [comment]
@nathanhammond what for?

**Comment by nathanhammond** [comment]
Just for me. 😛 If it doesn't show up I'll probably add it. Mostly so I can do: `ember new foo && cd foo && ember rm-welcome`. I create dozens of test apps for issues filed against CLI and want a fast workflow.

**Comment by ef4** [comment]
@nathanhammond it sounds like you really want `ember new --without-welcome`

Rails does this for a lot of pieces that people can opt out of if they want to go off the default path.

**Comment by nathanhammond** [comment]
Either way works but `--without-welcome` pushes the logic into Ember CLI's application blueprint and I'd rather not have to write custom removals for all of the default addons. I'd prefer delegating that to each addon individually. Otherwise it feels like we're playing whack-a-mole.

Instead if we improved the install hooks to have stories for "first installation" (runs on `ember new`) this would allow us to pull things out of the default blueprint. Then we could simply provide a list of npm packages to exclude from `package.json`.

But that's more work to solve for and accounts for some 15 cases. /complaining

**Comment by ef4** [comment]
I think it could be a general API. `--without-welcome-page` would drop the addon with namespace `welcome-page` from the default blueprint.

(I say "namespace" because that's already defined in the module unification RFC and it handles dropping default "ember-" and "ember-cli-" prefixes from npm module names.)

**Comment by nathanhammond** [comment]
Right, but being in the default blueprint in this case is _more_ than just an entry in `package.json`, it's also `{{welcome-page}}` in `application.hbs`. This is why it has to have some sort of custom hook to remove itself. We used to have default configuration for `ember-cli-qunit` in the app blueprint and there are probably other things still in there (he says, without looking).

Thinking about it more, though, `ember new` could probably safely run the default blueprint. This would mean an exclusion of the node package would succeed at the goal. I think exclusions at this point should be based on the npm package name, however, since the addon name isn't available until `npm install`ed and that defeats the point.

**Comment by acorncom** [comment]
> bundled command in the welcome-page addon to remove the welcome-page addon

@nathanhammond regarding the above, I think a lot of folks would like that, but we got ourselves in trouble the last time we mixed in a second feature to what was intended to be a fairly narrowly scoped change. I would suggest we just make the one change for now and revisit your suggestion later. Because we're changing something that **everyone** is affected by, it seems best to do these types of changes via an RFC. We had to get core heavily involved on this change because the *last* change wasn't done via an RFC. I'd rather not need to do that again :grinning:
